src/buildbot.py

```diff
@@ -38,7 +39,7 @@ def Fail(warn_only=False):
   """Mark one step as failing, but keep going."""
   sys.stdout.flush()
   if warn_only:
-    sys.stdout.write('\n@@@STEP_WARNINGS@@@\n')
+    Warn()
```

Comment:
It's a little bit nonobvious what `Warn()` means (i.e. print a warning) vs `Fail()` with a boolean flag.  I think it would be more clear if we kept `Fail()` just for hard failures, and then have `Warn()` do what `Fail(warn_only=True)` does today. Then they would be completely symmetric. 

Or,